canopy_const_mod.F90

```diff
@@ -0,0 +1,131 @@
+  REAL(RK),          PARAMETER     :: siday = 86164.09
+
+  ! mean gravitational acceleration [ m/sec**2 ]
+  ! --  mean of polar and equatorial values  (Source:  CRC76, pp. 14-6)
+  REAL(RK),          PARAMETER     :: grav = 9.80622
+
+  ! latitude degrees to meters
+!!!  REAL,          PARAMETER     :: dg2m = rearth * pi180
+  REAL(RK)                         :: dg2m
+
+  ! Solar Constant  [ W/m**2 ]  (Source:  CRC76, pp. 14-2)
+  REAL(RK),          PARAMETER     :: solcnst = 1373.0
+
+
+! Fundamental Constants: (Source: CRC76, pp. 1-1 to 1-6)
+
+  ! Avogadro's Constant [ number/mol ]
+  REAL(RK),          PARAMETER     :: avo = 6.0221367e23
+
+  ! universal gas constant [ J/mol-K ]
+  REAL(RK),          PARAMETER     :: rgasuniv = 8.314510
+
+  ! standard atmosphere [ Pa ]
+  REAL(RK),          PARAMETER     :: stdatmpa = 101325.0
+
+  ! standard temperature [ K ]
+  REAL(RK),          PARAMETER     :: stdtemp = 273.15
+
+  ! Stefan-Boltzmann [ W/(m**2 K**4) ]
+  REAL(RK),          PARAMETER     :: stfblz = 5.67051e-8
+
+
+! Non-MKS:
+
+  ! Molar volume at STP [ L/mol ] Non MKS units
+  REAL(RK),          PARAMETER     :: molvol = 22.41410
+
+
+! Atmospheric Constants: 
+
+  ! mean molecular weight for dry air [ g/mol ]
+  ! -- 78.06% N2, 21% O2, and 0.943% A on a mole fraction basis
+  !    (Source: Hobbs, 1995, pp. 69-70)
+  REAL(RK),          PARAMETER     :: mwair = 28.9628
+
+  ! dry-air gas constant [ 287.07548994 J/kg-K ]
+  REAL(RK),          PARAMETER     :: rdgas = 1.0e3 * rgasuniv / mwair
+
+  ! mean molecular weight for water vapor [ g/mol ]
+  REAL(RK),          PARAMETER     :: mwwat = 18.0153
+
+  ! gas constant for water vapor [ 461.52492604 J/kg-K ]
+  REAL(RK),          PARAMETER     :: rwvap = 1.0e3 * rgasuniv / mwwat
+
+  ! FSB NOTE: CPD, CVD, CPWVAP and CVWVAP are calculated assuming dry air and
+  ! water vapor are classical ideal gases, i.e. vibration does not contribute
+  ! to internal energy.
+
+  ! specific heat of dry air at constant pressure [ 1004.7642148 J/kg-K ]
+  REAL(RK),          PARAMETER     :: cpd = 7.0 * rdgas / 2.0
+
+  ! specific heat of dry air at constant volume [ 717.68872485 J/kg-K ]
+  REAL(RK),          PARAMETER     :: cvd = 5.0 * rdgas / 2.0
+
+  ! specific heat for water vapor at constant pressure [ 1846.0997042 J/kg-K ]
+  REAL(RK),          PARAMETER     :: cpwvap = 4.0 * rwvap
+
+  ! specific heat for water vapor at constant volume [ 1384.5747781 J/kg-K ]
+  REAL(RK),          PARAMETER     :: cvwvap = 3.0 * rwvap
+
+  ! vapor press of water at 0 C [ Pa ]  (Source: CRC76 pp. 6-15)
+  REAL(RK),          PARAMETER     :: vp0 = 611.29
+
+  ! The following values are taken from p. 641 of Stull (1988):
+
+  ! latent heat of vaporization of water at 0 C [ J/kg ]
+  REAL(RK),          PARAMETER     :: lv0 = 2.501e6
+
+  ! Rate of change of latent heat of vaporization w.r.t. temperature [ J/kg-K ]
+  REAL(RK),          PARAMETER     :: dlvdt = 2370.0
```

Comment:
At what temperature is this value valid? It is not perfectly linear.

Reply:
Good question.  Pulled this constant from another program, NACC/MCIP.  Need to check later.